File: 프로그래머스/1/132267. 콜라 문제/콜라 문제.java
class Solution {
    public int solution(int a, int b, int n) {
        int answer = 0;
        while(n >= a){
            int c = 0;
            c += n / a;
            answer += c * b;
            n = (c * b) + (n % a);
        }
        return answer;
    }
}

File: 프로그래머스/1/132267. 콜라 문제/콜라 문제.swift
import Foundation

func solution(_ a:Int, _ b:Int, _ n:Int) -> Int {
    var result = 0
    var empty = a
    var full = b
    var num = n
    while num >= empty {
        var cnt = 0
        cnt += num / empty
        result += cnt * full
        num = (cnt * full) + (num % empty)
    }
    return result
}